apps/costcontrol/test/unit/widget_startup_test.js

```diff
@@ -461,4 +453,94 @@ suite('Widget Startup Modes Test Suite >', function() {
     AirplaneModeHelper._status = 'disabled';
     Widget.init();
   });
+
+  test('DSDS start-up after SIM switching: from data usage only SIM to ' +
+         ' prepaid SIM',
+    function(done) {
+      /*
+        Related to:
+          Bug 1128493 - [CostControl] Widget broken after switching data sim
+      */
+
+      // SETUP TEST
+      var ftePending = false;
+      var self = this;
+      var fakeResult = {
+        datausage: requestDataUsageResult,
+        balance: resultRequestPrepaid
+      };
+      this.sinon.stub(SimManager, 'isMultiSim').returns(true);
+      this.sinon.stub(window.CostControl, 'getInstance')
+            .yields(newCostControlRequest(fakeResult));
+
+      function assertPrepaidDataLayout() {
+        done(function() {
+          var balanceDataText = _('currency', {
+            value: fakeResult.balance.data.balance,
+            currency: fakeResult.balance.data.currency
+          });
+          assertNonDataUseOnlyLayout('', balanceDataText);
+        });
+      }
+
+      function setupDataUsageOnlySIM() {
+        return new Promise(function(resolve) {
+          setupConfig('DATA_USAGE_ONLY', ftePending);
+          resolve();
+        });
+      }
+
+      function assertDataUsageOnlyLayout() {
+        return new Promise(function(resolve) {
+          window.addEventListener('hashchange', function _onHashChange(evt) {
```

Comment:
I'm thinking this is still configuration and we could move to the previous step or make a dedicated function in such a way, the `assert` function only asserts.